purchase_invoice_from_picking/tests/test_account_invoice.py

```diff
@@ -150,13 +153,27 @@ def test_invoice_filled_by_selecting_po__po_name_set_in_origin(self):
         assert self.order.name in self.invoice.origin
 
     def test_if_picking_selected_twice__no_duplicate_lines_added(self):
-        self.invoice.receipt_picking_id = self.picking
-        self.invoice.onchange_receipt_picking_id()
-        self.invoice.receipt_picking_id = self.picking
-        self.invoice.onchange_receipt_picking_id()
+        self.select_picking_on_invoice(self.picking, self.invoice)
+        self.select_picking_on_invoice(self.picking, self.invoice)
         lines = self.invoice.invoice_line_ids
         assert len(lines) == 2
 
+    def test_same_moves_not_added_on_different_invoices(self):
+        self.select_picking_on_invoice(self.picking, self.invoice)
+        self._remove_product_from_invoice(self.invoice, self.product_a)
+
+        invoice_2 = self.make_empty_invoice()
+        self.select_picking_on_invoice(self.picking, invoice_2)
+        assert len(invoice_2.invoice_line_ids) == 1
+        assert invoice_2.invoice_line_ids.product_id == self.product_a
+
+        invoice_3 = self.make_empty_invoice()
+        self.select_picking_on_invoice(self.picking, invoice_3)
+        assert not invoice_3.invoice_line_ids
+
+    def _remove_product_from_invoice(self, invoice, product):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/numigi/odoo-purchase-addons/pullRequest?prid=5371115)